### PR TITLE
Expose properties needed for octopus-psd

### PR DESCRIPTION
**Pull request by alexspevak**
In this pull request, we would like to expose needed properties on Layer and needed types and keys. Also, we added some functionality do be able to decode image patterns.

### CODE REVIEW

**Comment by scoiatael** [comment]
@pastelmind @dlehdanakf who could be a good reviewer here? Since I contributed part of the work to this PR I don't feel like I would be an impartial judge on that :)
Let me know if you need more background on why we want these features :)

**Comment by scoiatael** [comment]
@alexanderspevak let's convert it to a draft since it seems like there's more work to be done.

**Comment by alexspevak** [comment]
> @alexanderspevak let's convert it to a draft since it seems like there's more work to be done.

changed.

**Comment by alexspevak** [comment]
Hi there @pastelmind . Would you be so kind and tell us if we can merge this pull request, or if there are any changes that you would like to amend?

File: packages/psd/src/interfaces/AdditionalLayerInfo/PatternAliBlock.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// @webtoon/psd
+// Copyright 2021-present NAVER WEBTOON
+// MIT License
+
+import {ChannelBytes} from "../ChannelBytes";
+import {ChannelKind} from "../ChannelKind";
+import {KnownAliBlock} from "./AliBlockBase";
+import {AliKey} from "./AliKey";
+
+export interface PatterDataRectangle {
```
Comment (pastelmind) [inline packages/psd/src/interfaces/AdditionalLayerInfo/PatternAliBlock.ts:10]:
```suggestion
export interface PatterDataRectangle {
```

File: packages/psd/src/interfaces/AdditionalLayerInfo/PatternAliBlock.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  pixelDepth2: number;
+}
+
+export interface PatternData {
+  version: number;
+  length: number;
+  rectangle: PatterDataRectangle;
+  numberOfChannels: number;
+  channels: Map<ChannelKind, PatternDataChannel>;
+}
```
Comment (pastelmind) [inline packages/psd/src/interfaces/AdditionalLayerInfo/PatternAliBlock.ts:31]:
```suggestion
export interface PatternDataRectangle {
  top: number;
  left: number;
  bottom: number;
  right: number;
}

export interface PatternDataChannel extends ChannelBytes {
  written: true;
  length: number;
  pixelDepth1: number;
  rectangle: PatternDataRectangle;
  pixelDepth2: number;
}

export interface PatternData {
  version: number;
  length: number;
  rectangle: PatternDataRectangle;
  numberOfChannels: number;
  channels: Map<ChannelKind, PatternDataChannel>;
}
```

File: packages/psd/src/sections/LayerAndMaskInformation/AdditionalLayerInfo/readPatternAliBlock.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  const version = cursor.read("u32");
+  const length = cursor.read("u32");
+  const rectangle = readRectangle(cursor);
+  const numberOfChannels = cursor.read("u32");
+  const channels = new Map();
+
+  for (let i = 0; i < numberOfChannels + 2; i++) {
+    const channel = readChannel(cursor, height);
+    channels.set(i, channel);
+  }
```
Comment (pastelmind) [inline packages/psd/src/sections/LayerAndMaskInformation/AdditionalLayerInfo/readPatternAliBlock.ts:119]: Why make this a Map? Looks like an array would suffice.

**Review by pastelmind — CHANGES_REQUESTED** [review]
Hi. I apologize for the extreme delay in answering you. This PR looks good!

**Comment by pastelmind** [comment]
What if I created a separate PR based on this one, plus the proposed changes? Does this suit you?

**Comment by scoiatael** [comment]
Yeah, that works :)

File: packages/psd/src/sections/LayerAndMaskInformation/AdditionalLayerInfo/readPatternAliBlock.ts (hunk 3 above)
Comment (pastelmind) [reply packages/psd/src/sections/LayerAndMaskInformation/AdditionalLayerInfo/readPatternAliBlock.ts:119]: Upon examining the case further, it appears that the key for the Map is supposed to be `ChannelKind`, which can be one of 0, 1, 2, -1, -2, -3:

https://github.com/webtoon/psd/blob/f9ae80c23063b017f1d079d26a562b38ed84063d/packages/psd/src/interfaces/ChannelKind.ts#L7-L14

Perhaps using `ChannelKind` as the key was a mistake? In any case, I will change it to an array.

**Comment by pastelmind** [comment]
Closing in favor of #93

**Pull request closed without merging**